Makefile

```diff
@@ -33,62 +30,127 @@ else
 	mkdir -p ${BUILD_DIR}
 endif
 
-# Simple things, pointers into our build
-launcher: .pre-build
-	go run cmd/make/make.go -targets=launcher -linkstamp
+##
+## Build
+##
 
-table.ext: .pre-build
-	go run cmd/make/make.go -targets=table-extension -linkstamp
-table.ext-windows: .pre-build deps
-	go run cmd/make/make.go -targets=table-extension -linkstamp --os windows
+build_%: TARGET =  $(word 2, $(subst _, ,$@))
+build_%: OS = $(word 3, $(subst _, ,$@))
+build_%: OSARG = $(if $(OS), --os $(OS))
+build_%: ARCH = $(word 4, $(subst _, ,$@))
+build_%: ARCHARG = $(if $(ARCH), --arch $(ARCH))
+build_%: GOARG = $(if $(CROSSGOPATH), --go $(CROSSGOPATH))
+build_%: .pre-build
+	go run cmd/make/make.go -targets=$(TARGET) -linkstamp $(OSARG) $(ARCHARG) $(GOARG)
+
+fake_%: TARGET =  $(word 2, $(subst _, ,$@))
+fake_%: OS = $(word 3, $(subst _, ,$@))
+fake_%: OSARG = $(if $(OS), --os $(OS))
+fake_%: ARCH = $(word 4, $(subst _, ,$@))
+fake_%: ARCHARG = $(if $(ARCH), --arch $(ARCH))
+fake_%: .pre-build
+	go run cmd/make/make.go -targets=$(TARGET) -linkstamp -fakedata $(OSARG) $(ARCHARG)
+
+# The lipo command will combine things into universal
+# binaries. Because of the go path needs, there is little point in
+# abstracting this further
+lipo_%: build/darwin.amd64/% build/darwin.arm64/%
+	@mkdir -p build/darwin.universal
+	lipo -create $^ -output build/darwin.universal/$*
+
+# pointers, mostly for legacy reasons
+launcher: build_launcher
+tables.ext: build_tables.ext
+extension: build_osquery-extension.ext
+grpc.ext: build_grpc.ext
+fake-launcher: fake_launcher
 
 
-extension: .pre-build
-	go run cmd/make/make.go -targets=extension
+##
+## Cross Build targets
+##
 
-grpc-extension: .pre-build
-	go run cmd/make/make.go -targets=grpc-extension
+RELEASE_TARGETS=launcher osquery-extension.ext package-builder
+MANUAL_CROSS_OSES=darwin windows linux
+ARM64_OSES=darwin
+AMD64_OSES=darwin windows linux
 
+# xp is a helper for quick cross platform builds, and sanity checking
+# for breakage. humans only
+xp: $(foreach target, $(RELEASE_TARGETS), $(foreach os, $(MANUAL_CROSS_OSES), build_$(target)_$(os)))
 
-# Convenience tools
-osqueryi-tables: table.ext
-	osqueryd -S --allow-unsafe --verbose --extension ./build/darwin/tables.ext
-osqueryi-tables-linux: table.ext
-	osqueryd -S --allow-unsafe --verbose --extension ./build/linux/tables.ext
-osqueryi-tables-windows: table.ext
-	osqueryd.exe -S --allow-unsafe --verbose --extension .\build\windows\tables.exe
-sudo-osqueryi-tables: table.ext
-	sudo osqueryd -S --allow-unsafe --verbose --extension ./build/darwin/tables.ext
-launchas-osqueryi-tables: table.ext
-	sudo launchctl asuser 0 osqueryd -S --allow-unsafe --verbose --extension ./build/darwin/tables.ext
+# Actual release targets. Because of the m1 cgo cross stuff, this requires explicit go paths
+rel-amd64: CROSSGOPATH = /Users/seph/go1.15.6.darwin-amd64/bin/go
```

Comment:
Could this be `~/go1.14.6.darwin-amd64/bin/go` or does the path need to be that specific?

Reply:
It could probably use `~`. Once there's a real m1 release, I expect we'll drop these into nicer paths anyhow